# DP: recurse into xorgxrdp/ as well (make into proper configure option?)
# DP: also, actually install keymaps shipped (should be upstreamed)

--- a/Makefile.am
+++ b/Makefile.am
@@ -24,6 +24,7 @@ SUBDIRS = \
   mc \
   $(NEUTRINORDPDIR) \
   libxrdp \
+  librfxcodec \
   xrdp \
   sesman \
   keygen \
@@ -31,4 +32,5 @@ SUBDIRS = \
   instfiles \
   genkeymap \
   xrdpapi \
+  xorgxrdp \
   $(XRDPVRDIR)
--- a/common/Makefile.am
+++ b/common/Makefile.am
@@ -47,7 +47,12 @@ libcommon_la_SOURCES = \
   xrdp_rail.h \
   $(PIXMAN_SOURCES)
 
+if NEED_LIBDL
+EXTRA_LIBDL = -ldl
+endif
+
 libcommon_la_LIBADD = \
+  $(EXTRA_LIBDL) \
   -lcrypto \
   -lssl \
   -lpthread
--- a/configure.ac
+++ b/configure.ac
@@ -246,6 +246,14 @@ AC_CHECK_HEADER([X11/extensions/Xrandr.h
   [AC_MSG_ERROR([please install libxrandr-dev or libXrandr-devel])],
   [#include <X11/Xlib.h>])
 
+# checking if we need libdl
+save_LIBS=$LIBS
+AC_SEARCH_LIBS([dlopen], [dl dld], [], [
+  AC_MSG_ERROR([unable to find the dlopen() function])
+])
+AM_CONDITIONAL(NEED_LIBDL, [test x"$LIBS" != x"$save_LIBS"])
+LIBS=$save_LIBS
+
 CFLAGS="$save_CFLAGS"
 
 AC_SUBST([moduledir], '${libdir}/xrdp')
--- a/xrdpapi/Makefile.am
+++ b/xrdpapi/Makefile.am
@@ -4,6 +4,7 @@ EXTRA_LIBS =
 EXTRA_FLAGS =
 
 AM_CPPFLAGS = \
+  -I$(top_srcdir)/common \
   $(EXTRA_DEFINES) \
   $(EXTRA_INCLUDES)
 
@@ -18,4 +19,5 @@ libxrdpapi_la_LDFLAGS = \
   $(EXTRA_FLAGS)
 
 libxrdpapi_la_LIBADD = \
+  $(top_builddir)/common/libcommon.la \
   $(EXTRA_LIBS)
